### ISSUE TITLE
npm install failure about bcrypt

### ISSUE THREAD

**Issue by kreezxil** [kristianoye/kmud#1]
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.5 LTS
Release:        16.04
Codename:       xenial

debug log: http://paste.ubuntu.com/p/MG3Qjg8SxN/

**Comment by kristianoye** [comment]
I am trying to find the time to get back to this project, but it looks like bcrypt's version compatibility matrix is a bit wonky https://www.npmjs.com/package/bcrypt  I am currently running node v8.9 on Windows and v9.3 on CentOS Linux and bcrypt 1.0.3 works fine in both cases.  I stopped working on this project after discovering async functionality in node and I might go back and do some major rewrites in which case kmud will only work on later versions of node.

**Comment by LimpingNinja** [comment]
I'm currently working on nodejs@latest (12.8.0) and npm@latest (6.10.3) on Windows 10. I installed NodeJS via Scoop and Python27 via scoop (scoop add bucket versions) and simply did npm install bcrypt. The system starts up fine.

**Comment by LimpingNinja** [comment]
Note: If you remove node_modules and do the following:

'''
> npm install bcrypt --save
> npm install
'''

You should be good to go. @kristianoye - it's really technologically impressive. It's very rough on the lib side, but what you've odone is impressive. If you every start working on it again, let me know. I'd be happy to help.

**Comment by kristianoye** [comment]
I had a bunch of real life drop into my lap (bought a new house and am building out an add-on to my daughters bedroom atm).  I do really want to get back to this project, though.  I was going through and trying to clean up/streamline the "driver" before spending too much time on the lib (was in the process of making it all async).  I'll post back again when I can dedicate considerably more time to the project.

**Comment by kreezxil** [comment]
thanks for the update. side note, do you have a recommendation for a mud that has a coding structure similar to MUDOS or LPC that works on current Operating Systems? Such as Ubuntu 19 and Windows 10?

The client is not so much a big deal for me as there are so many alternatives available. I've been itching to restart an old mud concept I had. More like ressurect, because if you mudded during the 90's, I am Trimaster of Highlands I, II, and III and there were projects I had in those mudlibs that are still in my mind that I want to see come to life and then eventually port to Minecraft which is my new love. I still love muds.

**Comment by kreezxil** [comment]
when I say structure i'm talking about hot inject c-style code such as 
```c
#inherit "std/obj/monster"

void create() {
  setname("kristianoye");
  setshort("the coder of kmud");
  setpronoun("he");
  setpossessive("she's");
  setlong(
     "Before you stands Kristian\n" +
     getpronoun() +
     " has just bought a house and possibly an airplane.\n" +
     getpossessive() +
     " really busy. You're lucky " + getpossessive() + "here for you to look at."
  );
  sethp(100000);
  setstr(20);
  setdex(18);
  setchr(rnd(15)+15);
}

void init() {
   add_action("thank","thankFunc");
}

int thankFunc(string parseme) {
 write("You thank " + getname() + parseme + "\n");
 say(this_player()->get_name() + " thanks " + getname() + getshort() + parseme + "\n");
 return 1;
}
```

**Comment by LimpingNinja** [comment]
@kreezxil I can see if I can find it (but I think it is gone forever) - back in 2011 I made MUD engine using "Pike" scripting language which is a descendant of LPC. I aimed for efun compatibility but most of the efuns were not relevant, so there was still some tweaking required for code and a few changes to how this_player() worked (since it relied on backtrace and latter Pike versions borked it)

Correction, a search shows that this was 16 years ago. Dear god, where has time gone: https://pike-devel.lysator.liu.narkive.com/1kXiQ1AR/problems-with-backtrace

Currently I'm working in C# on a Mud Engine that uses Lua (MoonSharp) with some sandboxing as an embedded scripting language and an abstract/virtual file system. Mainly working on adding the proxy layer and bfuns (efuns) now, assuming 2 months I should have a basic Mudlib up and will open up the Github.

**Comment by kristianoye** [comment]
I experimented with Pike and DGD for awhile myself but never got into either.  I started my first MUD back in 1993 or so (Emerald MUD).  I was involved in developing a few others, too (Orion Delta, VargonMUD, and some others with names I can't remember).  I started playing on Muddog and Star Wars MUD.

The C# project sounds fun.  I thought about doing something similar with C# + ClearScript but the V8 engine doesn't do multithreading well which was my whole reason for wanting to use C#.  I am curious to see how Lua works for ya.

I am less concerned about supporting old school MUD clients and more interested in a web-based MUD and web-based tools (area builders, GUI editors, etc).  I am actively developing for both types of clients, though.

**Comment by kristianoye** [comment]
Also, @kreezxil , I do have a goal of creating a transpiler to convert LPC code like above into JavaScript but I haven't gotten around to it.  There are some things that do not translate well (e.g. multiple inheritance although even that can be sort of simulated)

**Comment by kreezxil** [comment]
At least there are 2 folks in this this thread got some interesting work
going on, let me know when you got something that needs testing.

**Comment by kristianoye** [comment]
I have merged in bcrypt 5.0.  I am going to close this issue.  I don't plan to support older versions of NodeJS moving forward.
